src/vanilla/store.ts

```diff
@@ -384,8 +384,13 @@ const buildStore = (getAtomState: StoreArgs[0]): Store => {
         return atomState
       }
     }
-    // Compute a new state for this atom.
-    atomState.d.clear()
+
+    if (isContinuablePromise(atomState.v) && atomState.v.status === PENDING) {
+      // If the atom has a pending promise, we should preserve the dependents
```

Comment:
I'm not sure if I understand the rationale behind it.
We are recomputing anyway, right? Is it because the "continuable promise" doesn't change its reference after the recomputation?

Reply:
in `2.8.4`, if the returned promise doesn't change reference, the dependents are not cleared.

Reply:
yeah, I mean, this is _kind of_ an intentional change in v2.9.0.